File: Python/Heap_PriorityQueue_Easy_LastStoneWeight_1046.py
import heapq

class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        heap = []

        for i in range(0, len(stones)):
            heapq.heappush(heap, -stones[i])

        while len(heap) > 1:
            max_1 = -heapq.heappop(heap)
            max_2 = -heapq.heappop(heap)

            if max_1 == max_2:
                continue
            else: 
                heapq.heappush(heap, -(abs(max_1 - max_2)))
        
        if len(heap) == 0:
            return 0
        else:
            return -heapq.heappop(heap)

File: C++/Heap_PriorityQueue_Easy_LastStoneWeight_1046.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) 
    {
        priority_queue<int> pq;

        for(int i = 0; i < stones.size(); i++)
        {
            pq.push(stones[i]);
        }    

        while(pq.size() > 1)
        {
            int max_1 = pq.top();
            pq.pop();
            int max_2 = pq.top();
            pq.pop();

            if(max_1 == max_2)
            {
                continue;
            }
            else
            {
                pq.push(abs(max_1 - max_2));
            }
        }

        if(pq.empty())
        {
            return 0;
        }
        else
        {
            return pq.top();
        }
        
    }
};